### ISSUE TITLE
Legends cut off in Jupyter Notebooks when plotting with %matplotlib notebook command

### ISSUE THREAD

**Issue by mtd91429** [matplotlib/matplotlib#9163]
### Bug report

When using matplotlib in a Jupyter Notebook with the magic `%matplotlib notebook`, legends placed outside of the axis are cut off. 

**Code for reproduction**

```python
import matplotlib.pyplot as plt
# probably unnecessary/overkill, but reset all parameters to default
plt.rcParams.update(plt.rcParamsDefault)
%matplotlib notebook

fig, ax = plt.subplots(ncols=1, nrows=1)
ax.plot(range(10), label="Plot 1")
ax.plot(range(10, 0, -1), label="Plot 2")
ax.legend(loc='center right', bbox_to_anchor=(1.20, 0.5))

```

**Actual outcome**

![image](https://user-images.githubusercontent.com/3360192/30137369-f9e7c966-9330-11e7-9e7a-efd1e60fd2c0.png)

**Expected outcome**

The legend would be fully displayed as in the following example. This plot obtained by switching to  `%matplotlib inline`
![image](https://user-images.githubusercontent.com/3360192/30137589-092d908a-9332-11e7-9c00-b9bc648e3ed6.png)
 

**Matplotlib version**

  * Operating System: Windows 7, x64
  * Matplotlib Version: 2.0.2
  * Python Version: 3.6.2
  * Jupyter Server: 5.0.0
  * Other Libraries: 

Matplotlib and python were installed using Anaconda

**Additional Information**
This seems similar to #2530

**Comment by jklymak** [comment]
The legend is cut off if you do `fig.savefig('Test.pdf')` as well.

`%matplotlib inline` is just letting you see off the edge of the figure, whereas `%matplotlib notebook` is giving you a more what-you-see-is-what-you-get view.  If you want yout legend over there, you can make the axis smaller (`fig.subplots_adjust=0.85`). 

An automatic soultion could be offered by `fig.tight_layout()`, but `tight_layout()` doesn't take into account legends (yet.  see https://github.com/matplotlib/matplotlib/issues/9130).

**Comment by jklymak** [comment]
OK, I made a pull request:  https://github.com/matplotlib/matplotlib/pull/9164

**Comment by timhoffm** [comment]
Closed by #9164.
